frameworks/TypeScript/bun/src/index.ts

```diff
@@ -0,0 +1,16 @@
+type HandlerFn = (req?: Request) => Promise<Response> | Response;
+
+const options: ResponseInit = { headers: { "Server": "Bun" } };
+
+const handlers = new Map<string, HandlerFn>();
+handlers.set("/json", () => Response.json({ message: "Hello, World!" }, options));
+handlers.set("/plaintext", () => new Response("Hello, World!", options));
+
+const server = Bun.serve({
+  port: 8080,
+  fetch(req: Request) {
+    return handlers.get(new URL(req.url).pathname)?.call(req);
```

Comment:
as a benchmark, I am not sure this it idiomatic or well performing + the `call(req)` when functions are arrows look off ..

```js
fetch: (req: Request) => handlers.get(new URL(req.url).pathname)?.()
```

looks already better, but I am not sure that's how people really write routes in Bun neither .. if it's idiomatic benchrmaks we're after, this PR doesn't look like the best we can do ... thoughts?

Reply:
I agree, actually I'm refactoring my PR to simplify this part and not implementing anything other than what is strictly necessary. Just plain `if` statements.